File: PrimeCalculator/无脑算法/Dart/main_optimized.dart
void main() {
  int limit = 1000000;
  int num = 2;
  int i;
  bool isPrime;

  while (num <= limit) {
    isPrime = true;
    if (num > 10 && (num % 2 == 0 || num % 10 == 5)) {
      isPrime = false;
    } else {
      i = 2;
      while (i * i <= num) {
        if ((num % i) == 0) {
          isPrime = false;
        }
        i++;
      }
    }
    if (isPrime) {
      print(num);
    }
    num++;
  }
}

File: PrimeCalculator/无脑算法/Python/main_optimized.py
# limit = int(input("Please enter a number for limit:"))
limit = 1000000

num = 2
while(num<=limit):
    isPrime = 1
    if num > 10 and (num % 2 == 0 or num % 10 == 5):
        isPrime = 0
    else:
        i = 2
        while(i*i<=num):
            temp = num % i
            if temp == 0:
                isPrime = 0
            i += 1
    if isPrime:
        print(num)
    num += 1
